website/versioned_docs/version-5.x/module-types.md

```diff
@@ -0,0 +1,51 @@
+---
+
+# Concept: single-spa microfrontend types
+
+Single-spa has [different categories](/docs/microfrontends-concept#types-of-microfrontends) of microfrontends. It is up to you where and how you use each of them. However, the single-spa core team has [recommendations](/docs/recommended-setup/#applications-versus-parcels-versus-utility-modules).
+
+Here is how each single-spa microfrontend works conceptually. This information should help you understand our [recommendations](/docs/recommended-setup/#applications-versus-parcels-versus-utility-modules).
+
+| Topic                | application                       | parcel                               | utility                              |
+| -------------------- | --------------------------------- | ------------------------------------ | ------------------------------------ |
+| routing              | has multiple routes               | no routes                            | no routes                            |
+| API                  | declarative API                   | imperative API                       | no single-spa api                    |
+| renders UI           | renders UI                        | renders UI                           | doesn't directly render UI           |
+| lifecycles           | single-spa managed lifecylces     | custom managed lifecycles            | no lifecycles                        |
+| When to use          | Core building block               | only needed with multiple frameworks | useful to share common logic         |
+
+Each single-spa microfrontend is an in-browser javascript module ([explaination](/docs/recommended-setup#in-browser-versus-build-time-modules)).
+
+## Applications
+
+### Applications are declarative
+Applications use a declarative api. Your single-spa config (also sometimes called root config) defines applications ahead of time, defines the condition at which they are active, but doesn't mount the application directly.
+
+### Applications have managed lifecycles
+single-spa manages registered applications and is in charge of all of their lifecycles. This prevents you from writing a bunch of logic about when applications should mount and unmount; single-spa takes care of that for you.
+All single spa needs to make this work automatically is for an activity function that describes when your application should be active.
+
+## Parcels
+
+### Parcels are imperative
+Parcels exist in many ways as an escape hatch from the normal declarative flow. They exist primarily to allow you to reuse UI across applications when those applications are written in multiple frameworks.
+
+### You manage the lifecycles of Parcels
+When you call `mountParcel` or `mountRootParcel` [(see api)](/docs/parcels-api) the parcel is mounted immediately and returns the parcel object. You need to call the `unmount` method on the parcel manually when the component that calls `mountParcel` unmounts.
+
+### Parcels are best suited for sharing UI between frameworks
```

Comment:
I think some people use parcels because they don't know what the alternative is. Should we perhaps have another subsection called "Alternatives to parcels" - maybe linking to the react styleguide that exports react components directly into other applications.

Reply:
That's a good idea. I don't have an example of that yet (but will soon) I created [an issue](https://github.com/single-spa/single-spa.js.org/issues/208) for it so I don't forget and put that in another MR